src/client.rs

```diff
@@ -13,96 +13,111 @@ use boardctrl::{PingRequest, PongRequest, ShutdownRequest, ShutdownReply, SendNo
 use std::env;
 use std::process;
 
+use clap::{Parser, Subcommand};
+
 pub mod boardctrl {
 	tonic::include_proto!("boardctrl");
 }
 
-#[tokio::main]
-async fn main() -> Result<(), Box<dyn std::error::Error>> {
+#[derive(Parser, Debug)]
+struct Cli {
+	/// The port of the connected server
+	#[clap(short, long, default_value = "50031")]
+	port: String,
+
+	#[clap(subcommand)]
+	command: Command,
+}
 
-	let command = env::args().nth(1).unwrap_or_else(|| {
-		println!("[CIVKIT-CLI] command required!");
-		process::exit(0x100);
-	});
+#[derive(Debug, Subcommand)]
+enum Command {
+	/// Send a ping message
+	Ping,
+	/// Shutdown the connected CivKit node
+	Shutdown,
+	/// Send a demo NIP-01 EVENT kind 1 to all the connected clients
+	Publishtextnote,
+	/// List information about connected clients [TODO]
+	Listclient,
+	/// List information about subscriptions [TODO]
+	Listsubscriptions,
+	/// Connect to a BOLT8 peer on local port
+	Connectpeer {
+		/// The port number for the peer
+		peer_local_port: String,
+	},
+	/// Disconnect from a client [TODO]
+	Disconnectclient,
+	/// Send a demo NIP-01 NOTICE to all the connected clients
```

Comment:
So currently there is no NIP specification about NIP-01 `NOTICE` type of message (“This NIP defines no rules for how `NOTICE` message should be sent or treated”). A future NIP could define “end of connection” (disconnecting the client) or “connection update” informing the client of service or link-level policy update (e.g bumping the satoshis per bytes for the subscriptions flows).

I think the “end of connection” could be a cleanup of the `MAGIC_SERVER_PAYLOAD` being a disconnection hack.